File: src/main/scala/zone/nora/simplestats/commands/SetKeyCommand.scala
package zone.nora.simplestats.commands

import java.io.File

import com.google.gson.JsonObject
import net.minecraft.command.{CommandBase, ICommandSender}
import org.apache.commons.io.FileUtils
import zone.nora.simplestats.SimpleStats
import zone.nora.simplestats.util.Utils

class SetKeyCommand extends CommandBase {

  override def getCommandName: String = "setkey"

  override def getCommandUsage(sender: ICommandSender): String = "/setkey [api-key]"

  override def processCommand(sender: ICommandSender, args: Array[String]): Unit = {
    val thread = new Thread(new Runnable {
      override def run(): Unit = {
        if (args.isEmpty) Utils.error("/setkey [api-key]", prefix = true) else {
          val key = args(0)
          if (key.length == 36 && Utils.validateKey(key)) {
            SimpleStats.key = key
            SimpleStats.valid = true

            val file = new File("config/simplestats.cfg")
            if (!file.exists()) file.createNewFile()

            val jsonObject = new JsonObject()
            jsonObject.addProperty("key", key)

            FileUtils.writeStringToFile(file, jsonObject.toString)
            Utils.put("\u00a7aYour API Key has been set.", prefix = true)
          } else Utils.error("This is not a valid Hypixel API key.", prefix = true)
        }
      }
    })
    thread.start()
  }

  override def canCommandSenderUseCommand(sender: ICommandSender): Boolean = true
}

File: src/main/kotlin/zone/nora/simplestats/commands/SetKeyCommand.kt
package zone.nora.simplestats.commands

import gg.essential.api.EssentialAPI
import gg.essential.api.utils.Multithreading
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender
import zone.nora.simplestats.config.Config
import zone.nora.simplestats.util.Utils

object SetKeyCommand : CommandBase() {
    override fun getCommandName(): String = "setkey"

    override fun getCommandUsage(sender: ICommandSender): String = "/setkey [api key]"

    override fun processCommand(sender: ICommandSender, args: Array<out String>) {
        if (args.size == 1) {
            val a0 = args[0]
            if (a0.lowercase() == "gui") {
                EssentialAPI.getGuiUtil().openScreen(Config.gui())
            } else {
                Utils.put("Setting API key.")
                Multithreading.runAsync {
                    Config.hypixelKeyS = a0
                    if (Config.validKey) {
                        Utils.put("Successfully set key!")
                    } else {
                        Utils.err("Set key... but it was invalid!!")
                        Utils.err("Double check that you input your key correctly.")
                    }
                    Config.markDirty()
                    Config.writeData()
                }
            }
        }
    }

    override fun canCommandSenderUseCommand(sender: ICommandSender): Boolean = true
}